xapian-letor/tests/api_letor.cc

```diff
@@ -1414,3 +1414,80 @@ DEFINE_TESTCASE(createfeaturevector_tfdoclencolllfcolllen, generated)
 
     return true;
 }
+namespace Xapian {
+
+class XAPIAN_VISIBILITY_DEFAULT CustomFeature : public Feature {
+  public:
+    CustomFeature() {
+	need_stat(TERM_FREQUENCY);
+	need_stat(DOCUMENT_LENGTH);
+	need_stat(COLLECTION_TERM_FREQ);
+	need_stat(COLLECTION_LENGTH);
+	need_stat(INVERSE_DOCUMENT_FREQUENCY);
+    }
+    std::vector<double> get_values() const {
+	return vector<double>();
+    }
+    std::string name() const {
+	return "CustomFeature";
+    }
+    void test_stats() {
+	// test for term frequency
+	TEST_EQUAL(get_termfreq("ZStiger"), 1);
+	TEST_EQUAL(this->get_termfreq("ZXDtiger"), 6);
+	TEST_EQUAL(this->get_termfreq("Ztiger"), 2);
```

Comment:
The `this->` qualification here isn't necessary (nor consistently applied!)

Some coding standards do say to use `this->` in some circumstances to make it clear the thing referred to is a member variable, but Xapian's doesn't.

Reply:
Oh, I forgot to remove them. fixed in a fixup commit.